File: packages/silo-client/src/request-options.ts
import type { CacheMode } from "./cache/cache-mode.js";

/**
 * The one cancellation convention on every call: the last parameter of
 * every method in the client. A per-call value wins over the client's own
 * default timeout.
 */
export interface RequestOptions {
  signal?: AbortSignal;
  timeoutMilliseconds?: number;
  /** `bypass` skips lookup and storage; `refresh` fetches and stores. Ignored on writes and disabled caches. */
  cache?: CacheMode;
}

File: packages/silo-client-dart/lib/src/request_options.dart
import 'cancellation_signal.dart';

/// The deadline and cancellation every call accepts last.
class RequestOptions {
  const RequestOptions({this.timeout, this.cancellation});

  /// Wins over `SiloOptions.timeout`.
  final Duration? timeout;
  final CancellationSignal? cancellation;
}
